ratelimit_var.go

```diff
@@ -0,0 +1,129 @@
+package goflags
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	stringsutil "github.com/projectdiscovery/utils/strings"
+	timeutil "github.com/projectdiscovery/utils/time"
+)
+
+type RateLimit struct {
+	MaxCount uint
+	Duration time.Duration
+}
+
+type RateLimitMap struct {
+	kv map[string]RateLimit
+}
+
+// Set inserts a value to the map. Format: key=value
+func (rateLimitMap *RateLimitMap) Set(value string) error {
+	if rateLimitMap.kv == nil {
+		rateLimitMap.kv = make(map[string]RateLimit)
+	}
+	var k, v string
+	if idxSep := strings.Index(value, kvSep); idxSep > 0 {
+		k = value[:idxSep]
+		v = value[idxSep+1:]
+	}
+	// note:
+	// - inserting multiple times the same key will override the previous value
+	// - empty string is legitimate value
+
+	if k != "" {
+		rateLimit, err := parseRateLimit(v)
+		if err != nil {
+			return err
+		}
+		rateLimitMap.kv[k] = rateLimit
+	}
+	return nil
+}
+
+// Del removes the specified key
+func (rateLimitMap *RateLimitMap) Del(key string) error {
+	if rateLimitMap.kv == nil {
+		return errors.New("empty runtime map")
+	}
+	delete(rateLimitMap.kv, key)
+	return nil
+}
+
+// IsEmpty specifies if the underlying map is empty
+func (rateLimitMap *RateLimitMap) IsEmpty() bool {
+	return rateLimitMap.kv == nil || len(rateLimitMap.kv) == 0
+}
+
+// AsMap returns the internal map as reference - changes are allowed
+func (rateLimitMap *RateLimitMap) AsMap() map[string]RateLimit {
+	return rateLimitMap.kv
+}
+
+func (rateLimitMap RateLimitMap) String() string {
+	defaultBuilder := &strings.Builder{}
+	defaultBuilder.WriteString("{")
+
+	var items string
+	for k, v := range rateLimitMap.kv {
+		items += fmt.Sprintf("\"%s\"=\"%s\"%s", k, v.Duration.String(), kvSep)
+	}
+	defaultBuilder.WriteString(stringsutil.TrimSuffixAny(items, ",", "="))
+	defaultBuilder.WriteString("}")
+	return defaultBuilder.String()
+}
+
```

Comment:
not sure if this method is required but if required i think we can just json encode map
current implementation does not seem to be accurate

```console
./basic -rls "h1=8/2h"
{"h1"="12h0m0s"}
map[h1:{8 12h0m0s}]
```

Reply:
This is required to implement flag.Value interface, so I modified it.